spi/zynq7000-spi/libzynq7000-spi.c

```diff
@@ -386,13 +401,18 @@ static int spi_initPins(spi_t *spi)
 	int err;
 
 	for (i = 0; i < sizeof(spi->pins) / sizeof(spi->pins[0]); i++) {
+		/* Skip not configured pins */
+		if (spi->pins[i].pin < 0) {
+			continue;
+		}
+
 		err = spi_setPin(spi->pins[i].pin, spi->pins[i].cfg);
 		if (err < 0) {
-			break;
+			return err;
```

Comment:
Why this change?

Reply:
If no SPI pin is to be configured (`spi->pins[i].pin < 0` for all pins) than the function returned uninitialzed `err`. Added this change to fix that issue.